app.py

```diff
@@ -47,16 +98,45 @@ def destroy_app(self):
 
     def create_panel(self):
         """
-        Shows the panel UI
+        Shows the UI as a panel. 
+        Note that since panels are singletons by nature,
+        calling this more than once will only result in one panel.
+        
+        :returns: The widget associated with the panel.
         """
         app_payload = self.import_module("app")
         
         # start the UI
         try:
-            return self.engine.show_panel(self._unique_panel_id, "Shotgun", self, app_payload.AppDialog)
+            widget = self.engine.show_panel(self._unique_panel_id, "Shotgun", self, app_payload.AppDialog)
         except AttributeError, e:
             # just to gracefully handle older engines and older cores
             self.log_warning("Could not execute show_panel method - please upgrade "
                              "to latest core and engine! Falling back on show_dialog. "
                              "Error: %s" % e)
-            return self.engine.show_dialog("Shotgun", self, app_payload.AppDialog)
+            widget = self.engine.show_dialog("Shotgun", self, app_payload.AppDialog)
```

Comment:
It feels to me like this could reuse `create_dialog` to keep the dialog creation logic isolated (e.g. setting `self._current_dialog`). Is there any reason why this couldn't be used here?

If changed to use `create_dialog`, any choice between doing a direct `return` in both cases or doing `widget = ...` looks good to me.